plugins/serializers/wavefront/wavefront.go

```diff
@@ -39,6 +41,12 @@ var tagValueReplacer = strings.NewReplacer("\"", "\\\"", "*", "-")
 
 var pathReplacer = strings.NewReplacer("_", ".")
 
+var bufPool = sync.Pool{
```

Comment:
I still think there is value to using a `sync.Pool` that is global to the package.  The `sync.Pool` will [pin](https://golang.org/src/sync/pool.go?#L178) the current goroutine to a pool, which should improve locality of reference.  Additionally, it does not make the `WavefrontSerializer` unsafe to use concurrently and is dynamic in size.

Reply:
There isn't any reason for it to be dynamic though because it will never need more than a single buffer.  Another issue with the pool is that you can't return the buffer to it unless you copy the data out of the buffer, I believe the current use in this pull request is unsafe even in a single threaded environment.

Reply:
> Another issue with the pool is that you can't return the buffer to it unless you copy the data out of the buffer

We do that.  The contents of pooled buffers are copied to the [`out`](https://github.com/influxdata/telegraf/blob/ec4b147bebacece7e6198d3ae783b3c9c8cea638/plugins/serializers/wavefront/wavefront.go#L95) buffer (note: `out` will never alias `buf`, [example](https://play.golang.org/p/lCmOVPwqrme)).  This code is thread-safe.

Personally, I would prefer to not add a mutable field to the `WavefrontSerializer`, but I understand the argument for simplicity and will make the change.

What are your thoughts on adding a mutex to protect the `WavefrontSerializer`'s scratch buffer?  From you comment and a quick scan of the code it appears there is no need for it to be thread-safe, but if that were to change it would be beneficial to have it as a guard against this code being overlooked during a refactor.

Reply:
I missed this, but this pool was added by accident the one we actually use is [`pbFree`](https://github.com/influxdata/telegraf/pull/5842/files#diff-cab4bf7f8e4b237a6576ccf08eea4f1cR197).  That said, I'm still removing the pools in favor of a shared buffer - as requested.

Reply:
Don't add anything to make it thread safe.  I agree it there is a risk of code sneaking in that use these wrong, especially with the way the SerializerOutput interface works (sending in a single instance), but I'd like to move the other way with parsers/serializers.

Reply:
Cool, I made `WavefrontSerializer` thread-safe.